### ISSUE TITLE
cc-wraper (i.e. when you use clang) broken on darwin

### ISSUE THREAD

**Issue by will** [NixOS/nixpkgs#261833]
### Describe the bug

I think https://github.com/NixOS/nixpkgs/pull/245282 broke `clang` on darwin.

The `mktemp --tmpdir cc-params.XXXXXX` command comes after `PATH="$path_backup"` which removes coreutils from the path. So the `mktemp` used here is the bsd one that ships with mac, which does not have `--tempdir`

Perhaps just `-t` would work, even though it's marked as deprecated in the manpage for the coreutils version. I don't know.

### Steps To Reproduce
1. nix run nixpkgs#clang

```
❯ nix run nixpkgs#clang
mktemp: illegal option -- -
usage: mktemp [-d] [-q] [-t prefix] [-u] template ...
       mktemp [-d] [-q] [-u] -t prefix
```

### Expected behavior

```
❯ clang
clang: error: no input files
```

### Notify maintainers

@reckenrode
@Ericson2314 
@amjoseph-nixpkgs

### Metadata

```console
❯ nix-shell -p nix-info --run "nix-info -m"
 - system: `"aarch64-darwin"`
 - host os: `Darwin 22.6.0, macOS 13.6`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.17.0`
 - channels(will): `"home-manager, nixpkgs, unstable"`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/Users/will/.nix-defexpr/channels/nixpkgs`
```

**Comment by reckenrode** [comment]
This is fixed by #258607.

**Comment by will** [comment]
Thanks! I assume it's still working it's way through the branches and just not yet on -unstable
